src/kubectlUtils.ts

```diff
@@ -1,344 +1,369 @@
-        ...Object.keys(env || {}).map((key) => `--env="${key}=${env[key]}"`)
-    ];
-
-    const runResult = await kubectl.invokeAsync(runCmd.join(" "));
-    if (runResult.code !== 0) {
-        throw new Error(`Failed to run the image "${image}" on Kubernetes: ${runResult.stderr}`);
-    }
-
-    return deploymentName;
-}
-
-/**
- * Query the pod list for the specified label.
- *
- * @param kubectl the kubectl client.
- * @param labelQuery the query label.
- * @return the pod list.
- */
-export async function findPodsByLabel(kubectl: Kubectl, labelQuery: string): Promise<KubernetesCollection<Pod>> {
-    const getResult = await kubectl.asJson<KubernetesCollection<Pod>>(`get pods -o json -l ${labelQuery}`);
-    if (failed(getResult)) {
-        throw new Error('Kubectl command failed: ' + getResult.error[0]);
-    }
-    return getResult.result;
-}
-
-/**
- * Wait and block until the specified pod's status becomes running.
- *
- * @param kubectl the kubectl client.
- * @param podName the pod name.
- */
-export async function waitForRunningPod(kubectl: Kubectl, podName: string): Promise<void> {
-    while (true) {
-        const shellResult = await kubectl.invokeAsync(`get pod/${podName} --no-headers`);
-        if (shellResult.code !== 0) {
-            throw new Error(`Failed to get pod status: ${shellResult.stderr}`);
-        }
-        const status = shellResult.stdout.split(/\s+/)[2];
-        kubeChannel.showOutput(`pod/${podName} status: ${status}`);
-        if (status === "Running") {
-            return;
-        } else if (!isTransientPodState(status)) {
-            const logsResult = await kubectl.invokeAsync(`logs pod/${podName}`);
-            kubeChannel.showOutput(`Failed to start the pod "${podName}". Its status is "${status}".
-                Pod logs:\n${logsResult.code === 0 ? logsResult.stdout : logsResult.stderr}`);
-            throw new Error(`Failed to start the pod "${podName}". Its status is "${status}".`);
-        }
-
-        await sleep(1000);
-    }
-}
-
-function isTransientPodState(status: string): boolean {
-    return status === "ContainerCreating" || status === "Pending" || status === "Succeeded";
-}
-
-/**
- * Get the specified resource information.
- *
- * @param kubectl the kubectl client.
- * @param resourceId the resource id.
- * @return the result as a json object, or undefined if errors happen.
- */
-export async function getResourceAsJson<T extends KubernetesResource | KubernetesCollection<KubernetesResource>>(kubectl: Kubectl, resourceId: string, resourceNamespace?: string): Promise<T | undefined> {
-    const nsarg = resourceNamespace ? `--namespace ${resourceNamespace}` : '';
-    const shellResult = await kubectl.asJson<T>(`get ${resourceId} ${nsarg} -o json`);
-    if (failed(shellResult)) {
-        vscode.window.showErrorMessage(shellResult.error[0]);
-        return undefined;
-    }
-    return shellResult.result;
-}
+import * as vscode from "vscode";
```

Comment:
I'm seeing whitespace diffs on every line of this file... any idea what's causing those?  Would be nice to have only the functional diffs in this PR, and if our whitespace is messed up then address that in a separate non-functional PR.

Reply:
My editor replaced \r\n with \n, it's kind of hard to revert, is it required?

Reply:
actually had to fix this to be able to rebase :)